x/consensus/keeper/keeper.go

```diff
@@ -3,60 +3,76 @@ package keeper
 import (
 	"context"
 
-	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
-
+	"cosmossdk.io/collections"
 	storetypes "cosmossdk.io/core/store"
+	"cosmossdk.io/errors"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	cmttypes "github.com/cometbft/cometbft/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 
 	"github.com/cosmos/cosmos-sdk/codec"
-	"github.com/cosmos/cosmos-sdk/x/consensus/exported"
 	"github.com/cosmos/cosmos-sdk/x/consensus/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
 )
 
-var _ exported.ConsensusParamSetter = (*Keeper)(nil)
+var (
+	// ParamsPrefix = collections.NewPrefix([]byte{})
+	ParamsPrefix = collections.NewPrefix("Consensus")
+	StoreKey     = "Consensus"
+)
 
 type Keeper struct {
 	storeService storetypes.KVStoreService
-	cdc          codec.BinaryCodec
 
 	authority string
+	Params    collections.Item[cmtproto.ConsensusParams]
 }
 
 func NewKeeper(cdc codec.BinaryCodec, storeService storetypes.KVStoreService, authority string) Keeper {
+	sb := collections.NewSchemaBuilder(storeService)
 	return Keeper{
 		storeService: storeService,
-		cdc:          cdc,
 		authority:    authority,
+		Params:       collections.NewItem(sb, ParamsPrefix, "Consensus", codec.CollValue[cmtproto.ConsensusParams](cdc)),
```

Comment:
i think instead of using `Consensus` we should define what the collection represents which are `params`.